plugins/out_stackdriver/stackdriver_conf.c

```diff
@@ -281,6 +308,32 @@ struct flb_stackdriver *flb_stackdriver_conf_create(struct flb_output_instance *
         ctx->resource = flb_sds_create(FLB_SDS_RESOURCE_TYPE);
     }
 
+    flb_kv_init(&ctx->resource_labels);
+
+    /*
+     * Load monitored resource labels from config files.
+     * Labels that don't match for a particular resource type will be
+     * rejected by the API.
+     *
+     * For a list of valid labels, see:
+     * https://cloud.google.com/logging/docs/api/v2/resource-list
+    */
+    mk_list_foreach(labels, &ins->properties) {
+        kv = mk_list_entry(labels, struct flb_kv, _head);
+        if (strncasecmp(kv->key, rlabel_prefix, rlabel_prefix_len) == 0 &&
+            flb_sds_len(kv->key) > rlabel_prefix_len) {
+                /* Copy actual label back to kv key */
+                kv->key = flb_sds_create(kv->key + rlabel_prefix_len);
+                /* Create a new label value pair in list */
+                flb_kv_item_create(&ctx->resource_labels, kv->key, kv->val);
+            }
+    }
+
+    tmp = flb_kv_get_key_value("project_id", &ctx->resource_labels);
+    if (!tmp && ctx->project_id) {
+        flb_kv_item_create(&ctx->resource_labels, "project_id", ctx->project_id);
+    }
```

Comment:
So, this allows the user to override the project id from the credentials with the label in the config. Probably worth adding a comment to that effect (and documenting in the README at some point).

Reply:
I am going to remove the ability to override project_id, it doesn't make much sense since the credentials provide it.

Reply:
After talking, I decided to not remove this ability.